### PR TITLE
Add API for content fetching

### PR DESCRIPTION
**Pull request by marcospassos**
## Summary
Add API for content fetching and the preview widget.

### Dependency
- https://github.com/croct-tech/sdk-js/pull/276

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/deploy-published-releases.yaml [hunk 1]
```diff
@@ -6,9 +6,12 @@ on:
       - published
 
 env:
+  # todo: review these URLs
```
Comment (marcospassos) [inline .github/workflows/deploy-published-releases.yaml:9]: @Fryuni, we need to discuss this URL hierarchy

File: preview/.gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+node_modules/
+dist/
+.parcel-cache/
+/test-results/
+/playwright-report/
+/playwright/.cache/
+/test-results/
+/playwright-report/
+/playwright/.cache/
```
Comment (marcospassos) [inline preview/.gitignore:9]: duplicated

File: preview/src/index.html [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            width: 100%;
+            height: calc(2 * 100vh);
+            background-color: #F6F7F8;
+        }
+    </style>
+</head>
+<body>
+<iframe id="widget" src="./widget.html"></iframe>
+
+<script type="text/javascript">
```
Comment (marcospassos) [inline preview/src/index.html:19]: Simplify this implementation. This is for development purpose only.

File: tsconfig.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
     "outDir": "build",
     "downlevelIteration": true,
     "declaration": true,
-    "lib": ["es6", "dom", "dom.iterable", "es2018.promise"]
+    "lib": ["es6", "dom", "dom.iterable", "es2018.promise", "ES2019.Object"]
```
Comment (Fryuni) [inline tsconfig.json:20]: Do we still need to target ES5? Only Opera Mini still doesn't support ES6.
In fact, the coverage of [`es2018.promise` support is over 96%](https://caniuse.com/promise-finally)

These polyfills are a huge portion of our bundle size. Should we keep them?

**Review by Fryuni — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 4 above)
Comment (marcospassos) [reply tsconfig.json]: not at all

**Review by Fryuni — APPROVED** [review]

**Pull request merged**